### PR TITLE
Backport epub generator for Rails 7.0

### PR DESCRIPTION
**Pull request by hsbt**
### Motivation / Background

https://guides.rubyonrails.org/ still provides mobi format for Kindle platform. But mobi format is already EOL in Dec, 2022.

### Detail

We should backport https://github.com/rails/rails/pull/45539 to `7-0-stable` branch.

### Additional information

I'm not sure the detail of guide publishing workflow. Is it enough changes?

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [ ] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [ ] Tests are added or updated if you fix a bug or add a feature.
* [ ] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

**Comment by skipkayhil** [comment]
> I'm not sure the detail of guide publishing workflow. Is it enough changes?

From what I understand, new versions of guides are published on release, so the change would show up when 7.0.5 is released (whenever that is)

**Comment by hsbt** [comment]
https://buildkite.com/rails/rails/builds/91864#0185328f-d3ae-4f0f-9e6a-692533b194a6 seems fragile result. Does anyone restart this?

**Comment by eileencodes** [comment]
Thanks for the backport! The one CI failure was unrelated to these changes so I merged.

**Comment by hsbt** [comment]
@eileencodes Thanks ❤️

File: guides/rails_guides/generator.rb [hunk 1]
```diff
@@ -146,14 +139,21 @@ def generate?(source_file, output_file)
...
         output_path = output_path_for(output_file)
         puts "Generating #{guide} as #{output_file}"
-        layout = @kindle ? "kindle/layout" : "layout"
+        layout = @epub ? "epub/layout" : "layout"
 
         view = ActionView::Base.with_empty_template_cache.with_view_paths(
           [@source_dir],
           edge:     @edge,
           version:  @version,
+<<<<<<< HEAD
```
Comment (vtamara) [inline guides/rails_guides/generator.rb:148]: I don't think this should be here.

Comment (eileencodes) [reply guides/rails_guides/generator.rb:148]: Thanks! Fixed in 6f2f0a7e7103e23abe082c07c430ddda425f5e97

Comment (hsbt) [reply guides/rails_guides/generator.rb:148]: Ah, sorry, and thanks!

**Pull request merged**